### DISCUSSION TITLE
Need Help with pushing my game

### DISCUSSION THREAD

**Discussion by Yousuf-Sh** [community/community#31851] [New to GitHub] [answered]
Hello, Mr. Github, 
i had been working on the prototype of a game on unity for a few weeks, but no i want to bring a friend on to the prroject so that we both can work on it.
i have created a repo, and added and commited the project folder to it via the desktop app, when i try to push it , it goes up to ~90% on the "wirting objects" step and the exits with "remote disconnected, make sure your internet connection is stable"
from what i have researched, up until now, it has something to do with the overall size of the push, the original project folder is ~2GB, i do understand the i have to integrate LFS into my repo, but all the tutorials that i have watched are talking about single file problems, i can not for the life of me figure out what i need to do in order to just get all of the files in my unity project folder on to the  github repo so that me and my friend can start collaborating on the game.
Thanks

**Comment by melvincwng** [comment, marked as answer]
@Yousuf-Sh, since you've mentioned that your git push command to the remote repository is failing due to large file sizes, here are my suggestions:

1. Instead of one large commit (containing many large files), try splitting up the original commit into two or three commits - with each commit containing just some files.
2. Example - one commit can have 50% of the files, and the other commit can have the remaining 50% of the files required to be pushed
3. If that still doesn't work, try splitting your files into more commits (e.g. 3 or 4 or 5 etc)

Hopefully, the splitting of files into smaller commits will prevent this LFS issue when pushing to GitHub's remote servers.

Let's hope this suggestion works and thus answers your question 🙏🏻.

Good luck!

**Reply by Yousuf-Sh** [reply]
Thanks, this worked, had to split the push into 17 parts tho.

**Reply by melvincwng** [reply]
@Yousuf-Sh , glad this worked! Pleasure to have helped you 😄

**Reply by Boysin-balde** [reply]
> @Yousuf-Sh, since you've mentioned that your git push command to the remote repository is failing due to large file sizes, here are my suggestions:
> 
> 1. Instead of one large commit (containing many large files), try splitting up the original commit into two or three commits - with each commit containing just some files.
> 2. Example - one commit can have 50% of the files, and the other commit can have the remaining 50% of the files required to be pushed
> 3. If that still doesn't work, try splitting your files into more commits (e.g. 3 or 4 or 5 etc)
> 
> Hopefully, the splitting of files into smaller commits will prevent this LFS issue when pushing to GitHub's remote servers.
> 
> Let's hope this suggestion works and thus answers your question 🙏🏻.
> 
> Good luck!



> Thanks, this worked, had to split the push into 17 parts tho.

I had this kind of issues and I tried splitting the files up but It didn't worked Can you suggest other solutions

**Reply by Yousuf-Sh** [reply]
@Boysin-balde can you please give me a bit more detail about what you tried?

**Reply by Yousuf-Sh** [reply]
@Boysin-balde  also have read of My reply at https://github.com/community/community/discussions/31851#discussioncomment-3717603
maybe this can help

**Reply by Boysin-balde** [reply]
> @Boysin-balde also have read of My reply at [#31851 (reply in thread)](https://github.com/orgs/community/discussions/31851#discussioncomment-3717603) maybe this can help

Ok thanks for your help

**Comment by shahziabbhatti** [comment]
Hi, Mr. Github,
I had been working on the model of a game on solidarity for half a month, yet no I need to welcome a companion on to the prroject so we both can deal with it.
I have made a repo, and added and commited the venture envelope to it through the work area application, when I attempt to push it , it increases by to ~90% on the "wirting objects" step and the ways out with "remote detached, ensure your web association is steady"
from what I have explored, as of not long ago, it has something [astroluna ](https://astroluna.org/)to do with the general size of the push, the first undertaking organizer is ~2GB, I truly do comprehend the I need to coordinate LFS into my repo, yet every one of the instructional exercises that I have watched are discussing single record issues, I can not for the existence of me sort out what I really want to do to simply get each of the documents in my solidarity project envelope on to the github repo so me and my companion can begin teaming up on the game.
Much appreciated.

**Reply by Yousuf-Sh** [reply]
your commit size meeds to be less than 100 files, and smaller than some certain file size i think (50MB) (this is per commit/push limit doesn't mean your entire project needs to be smaller size than this number) You dont need LFS. i was able to push a whole unity project by just commiting and pushing ~30 files at a time.
In order to make sure you are only pushing the needed files, make sure that your repository  has an appropriate ".gitignore" file.
when creating  your repo, you will get the option to add one at the bottom  of the creation menu. if you are using unity, or unreal engine, just add it's appropriate gitIgnore file from the dropdown list. if yyou sre using some other engine im sure its gitignore file also exosts, you will have to look for it.
Git ignore file excludes any unwanted log files or meta files or asset store files that arent needed. 
make sure that the git ignore file is located inside the root of your project. 
for me it was able to reduce the file count from 3000 to ~340. i had to do around 11 commits and pushes before everything was on my remote.

**TLDR:** commit and push a smaller number of files at a time, include a .gitignore file in your repo.
Hope this helps!
